packages/xo-web/src/xo-app/settings/audit/index.js

```diff
@@ -124,17 +123,18 @@ const openFingerprintPromptModal = () =>
         <Icon icon='diagnosis' /> {_('auditCheckIntegrity')}
       </span>
     ),
-  }).then((value = '') => {
-    value = value.trim()
-    return value !== '' ? value : DEFAULT_HASH
-  }, noop)
+  }).then((value = '') => value.trim(), noop)
 
 const checkIntegrity = async () => {
   const fingerprint = await openFingerprintPromptModal()
   if (fingerprint === undefined) {
     return
   }
 
+  if (fingerprint === '') {
+    return openGeneratedFingerprintModal(await generateAuditFingerprint())
+  }
```

Comment:
IMO, to make it look like a clear chronological workflow, this should stay at the end of the function and the `if` should be around the `checkAuditRecordsIntegrity` step instead.